### PR TITLE
Add /config handler which exposes the current YAML config of cortex.

### LINKED ISSUE #1886: Expose HTTP endpoint to export live config in yaml format
Given the intention to deprecate CLI flags, I'm wondering if it's useful / required to give users a way to convert CLI flags to yaml. An option - for users already running Cortex clusters - may be introduce an HTTP endpoint to export the live config (which is the result of yaml + CLI flags + defaults) as yaml.

_I'm not super sure this is useful, so I would like to gather some feedback and real use cases first._

### PR DESCRIPTION
**Pull request by tomwilkie**
Signed-off-by: Tom Wilkie <tom@grafana.com>

Fixes #1886

**Checklist**
- [ ] Tests updated
- [x] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Comment by cyriltovena** [comment]
> I think we need to mask passwords.

Good point, although I think the usage is meant to be internal.

File: CHANGELOG.md [hunk 1]
```diff
@@ -19,6 +19,7 @@
 * [FEATURE] Added sharding support to compactor when using the experimental TSDB blocks storage. #2113
 * [FEATURE] Add ability to override YAML config file settings using environment variables. #2147
   * `-config.expand-env`
+* [FEATURE] Add /config HTTP endpoint which exposes the current Cortex configuration as YAML.
```
Comment (pracucci) [inline CHANGELOG.md]:
```suggestion
* [FEATURE] Add /config HTTP endpoint which exposes the current Cortex configuration as YAML. #2165
```

We now a script to check the changelog when we do a release, and it's based on the PR number in the changelog entry, reason why it's important to always add it.

Comment (tomwilkie) [reply CHANGELOG.md]: Good to know, thanks!

**Comment by tomwilkie** [comment]
Thanks @pracucci!  Your commits LGTM.  Good catch re:omitempty, yes I think these should all be removed - and there are a lot of them.

**Pull request merged**